### ISSUE TITLE
Need Help - Device Offline 

### ISSUE THREAD

**Issue by srikumarc** [blynkkk/beta#149]
On new BLYNK IOT app:
1. I could create a template and a device.
2. Could flash the code to my Node mcu device.
3. on the App in my mobile phone the new device shows as "offline". How can I flash the network credentials in such situations?
4. On the web app also it shows offline.
5. I can see the local hotspot of the device but when I connect to it the Mobile app it  goes off line and messages that it is unable to connect to server

I deleted the device and template and created a new one and went through the whole process but resulting in the same issue

**Comment by doom369** [comment]
@srikumarc hello, have you checked the docs - https://docs.blynk.io/en/getting-started/activating-devices/blynk-edent-wifi-provisioning#setting-up-your-sketch-to-enable-wifi-provisioning?

**Comment by srikumarc** [comment]
Thanks. I did everything according to the document. I am now stuck with the problem of activating the device from mobile app. 

First Attempt : 

The mobile app detects the device during scan and when I select it, goes through the three steps1. Configuring wifi, 2. Reconnecting back to cloud but the third step, The first two steps are fine but with step 3. waiting for device online does not complete even after long time (30 min). 

Second Attempt: Changed to a new Node MCU, reflashed the code but the result is the same as above. Gets stuck at step # 3.

Third Attempt:

Went back to the previous Node MCU, reflashed the code  but this time it detects the Access point but does not connect to it. Tried Manually also the same result. Obviously something is wrong at my end.

**Comment by srikumarc** [comment]
Further to the previous message, I tried performing the above steps using my iPhone and it stops at waiting for device online. I tried on two different wifi networks with the same result

**Comment by doom369** [comment]
@srikumarc do you use the latest firmware? Did you provide correct wi-fi settings?

**Comment by srikumarc** [comment]
yes. I tried three firmwares today and strangely all three were working
fine the day before.

The iphone Blynk app is more user friendly and with that I know that it is
going through the various stages.

1. sometimes it says cant connect to device.
2. sometimes it goes through to getting device info and then says cant
connect to device.
3. and sometimes it goes further and completes wifi configuration and then
waits for device...nothing happens after that.

best regards

srikumar

**Comment by srikumarc** [comment]
I forgot to add the following to item 3 in my previous message.....

after waiting for device it quits with the message "configuring error"

**Comment by doom369** [comment]
@srikumarc are you able to view/show the sertial output of the device?

**Comment by srikumarc** [comment]
Yes, the serial monitor does not always show but after your message I tried
flashed the firmware again.

Thid time it I'd showing upto line -

CONFIGURING = SWITCH_TO_ STA

but the device doesnt connect

best regards

srikumar

**Comment by srikumarc** [comment]
Please see the serial monitor output attached

Srikumar

**Comment by srikumarc** [comment]
[Serial monitor output.pdf](https://github.com/blynkkk/blynk_Issues/files/6726818/Serial.monitor.output.pdf)
[Serial monitor output.pdf](https://github.com/blynkkk/blynk_Issues/files/6727412/Serial.monitor.output.pdf)

**Comment by vshymanskyy** [comment]
1. You should use the latest version of library, not the beta.
2. Check your ESP core version, update to latest if possible.
3. Check if your wifi credentials are correct, and you have good wifi signal (say, 5 meters from the router for this initial test)

**Comment by srikumarc** [comment]
Thanks volodymyr. I tried the following as per your suggestion.

a. Deleted the Beta libraries and went back to version 1.0.0
b. My ESP core is the latest version
c. My wifi credentials were correct because I tested it just to be sure on
my mobile phone

It did not work with the above changes and the device wouldn't go online
and was getting stuck at the same step as in the serial monitor output I
had sent you earlier and on the app it was waiting at step 2. "awaiting
device"

Next I tried a trick,

a. I opened the basic example code for Blynk edgent - ESP 8266 and the bare
code and edited just  only the header codes for the template/device/board
and uploaded the code and the device went online !
b. Next I copied the rest of my code and uploaded again and this time the
device remained online and it is now working !

Obviously there is some bug somewhere. I hope you can look into it and fix
it

I find the Android app not at all useful to add a device because it just
doesn't show the steps like the IOS app and this is an area you must fix.

In general my experience with the new Blynk app has been less than smooth !
. unless the app can take on board people with basic knowledge such as
myself it won't be a success. Please do not call the old Blynk App a
"legacy" app and I would rather prefer that it stays as Blynk "lite" or
something like that and the new app called Blynk  "Pro".

I have to convert a few of my other devices to the new app and hope I will
not face any problems.

One area where I need clear step by step instruction on is how can I change
the wifi credentials at the site of device installation? I hope I will not
face problems again when I do it.

Best regards

Srikumar

**Comment by orilzzz** [comment]
I had the same problem while using my "ESP32 devkit C v4" board. After going through the entire internet, I was able to solve my problem by downgrading my esp32 board, to version 1.0.0 in the Board Manager of the Arduino IDE. 

Really terrible and time-consuming experience. Hopefully, this glitch is resolved soon, and there is proper documentation.

**Comment by ShahFarhan548** [comment]
@vshymanskyy i am going through the same issue, i can see in my network connection that my device is connected to the internet but its status is still shown as offline, i tried it multiple times still unable to fix the issue its time consuming. if some one got the solution please let me know
